### DISCUSSION TITLE
MQTT Suggestion - central MQTT access data

### DISCUSSION THREAD

**Discussion by thomaswitt** [koush/scrypted#1097] [Ideas]
UI wise it is a little annoying is that I have to re-enter the access data in every single device.

In an ideal world, I could configure a preset in the MQTT plugin, as well as a prefix for the topic which gets autocomposed so that a device called “Cam Entry” will become with the prefix “mqtt://192.168.1.1/scrypted/” automatically “mqtt://192.168.1.1/scrypted/cam-entry” by default (which you can overwrite if you like).

Right now I have to go into every sensor, click on MQTT and enter the login date over and over again.

**Comment by brandon-fryslie** [comment]
Yes running into this myself.  Will see if I can get a dev env set up and submit a PR.  I'm glad the functionality is there, the hard work is done.  But it's missing the last 10% to make it good.
